airbyte-integrations/connectors/source-chargebee/source_chargebee/source.py

```diff
@@ -0,0 +1,205 @@
+        }
+        super().__init__()
+
+    @backoff.on_exception(
+        backoff.expo,  # Exponential back-off
+        OperationFailedError,  # Only on Chargebee's OperationFailedError
+        max_tries=MAX_TRIES,
+        max_time=MAX_TIME,
+    )
+    def _send_request(self) -> ListResult:
+        """
+        Just a wrapper to allow @backoff decorator
+        Reference: https://apidocs.chargebee.com/docs/api/#error_codes_list
+        """
+        # From Chargebee
+        # Link: https://apidocs.chargebee.com/docs/api/#api_rate_limits
+        list_result = self.api.list(self.params)
+        return list_result
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        """
+        Override airbyte_cdk Stream's read_records method
+        """
+        # Add offset to params if found
+        # Reference for Chargebee's pagination strategy below:
+        # https://apidocs.chargebee.com/docs/api/#pagination_and_filtering
+        pagination_completed = False
+        if stream_state:
+            self.params.update(stream_state)
+        # Loop until pagination is completed
+        while not pagination_completed:
+            # Request the ListResult object from Chargebee
+            # with back-off implemented through self._send_request()
+            list_result = self._send_request()
+            # Read message from results
+            for message in list_result:
+                yield message._response[self.name]
+            # Get next page token
+            self.next_offset = list_result.next_offset
+            if self.next_offset:
+                self.params.update({"offset": self.next_offset})
+            else:
+                pagination_completed = True
+
+        # Always return an empty generator just in case no records were ever yielded
+        yield from []
+
+    def get_updated_state(
+        self,
+        current_stream_state: MutableMapping[str, Any],
+        latest_record: Mapping[str, Any],
+    ):
+        """
+        Override airbyte_cdk Stream's get_updated_state method
+        to get the latest Chargebee stream state
+        """
+        # Init the current_stream_state
+        current_stream_state = current_stream_state or {}
+
+        # Get current timestamp
+        # so Stream will sync all records
+        # that have been updated before now
+        now = pendulum.now().int_timestamp
+        current_stream_state.update(
+            {
+                "update_at[before]": now,
+            }
+        )
+
+        # Get the updated_at field from the latest record
+        # using Chargebee's Model class
+        # so Stream will sync all records
+        # that have been updated since then
+        print(latest_record)
```

Comment:
we should remove this print statement otherwise it'll be a pretty big log file :) 
```suggestion
```